Mathlib/Computability/Primrec.lean

```diff
@@ -22,6 +22,19 @@ we need that the composition of encode with decode yields a
 primitive recursive function, so we have the `Primcodable` type class
 for this.)
 
+In the above, the pairing function is primitive recursive by definition.
+This deviates from the textbook definition of primitive recursive functions,
+which instead work with *`n`-ary* functions. We formalize the textbook
+definition in `Nat.Primrec'`. `Nat.Primrec'.prim_iff` then proves it is
+equivalent to our chosen formulation.
```

Comment:
You could also mention that this design choice is discussed in more detail in [carneiro2019], referenced below

Reply:
More generally, you mention "I have found some parts of the Computability development hard to follow without referencing the linked paper.", which is understandable, but I think it's valuable to explicitly suggest to the reader that that paper might be helpful to read. In particular, many references in mathlib are for an informal account of the mathematics, whereas this one is an account of the formal mathematics, so a little extra attention to it is warranted.